geth/swarm.go

```diff
@@ -0,0 +1,186 @@
+package geth
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/contracts/chequebook"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/logger"
+	"github.com/ethereum/go-ethereum/logger/glog"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/swarm"
+	bzzapi "github.com/ethereum/go-ethereum/swarm/api"
+	"path/filepath"
+	"sync"
+)
+
+var (
+	ErrInvalidSwarmService           = errors.New("swarm service is unavailable")
+	ErrSwarmIdentityInjectionFailure = errors.New("failed to inject identity into Swarm")
+)
+
+type SwarmService struct {
+	swarm  *swarm.Swarm
+	keys   map[string]*ecdsa.PrivateKey
+	keysMu sync.RWMutex // Mutex to sync identity keys
+}
+
+func NewSwarmService() *SwarmService {
+	service := &SwarmService{
+		keys: make(map[string]*ecdsa.PrivateKey),
+	}
+
+	return service
+}
+
+// activateEthService configures and registers the eth.Ethereum service with a given node.
```

Comment:
- fix comment
- consider renaming method, for ex `RunSwarmNode(pkey)` or sth, otherwise `swarm.Start()` and `swarm.StartSwarm()` are a bit confusing